plugin/src/init.server.lua

```diff
@@ -1,222 +1,278 @@
-	return not not table.find(Settings.include, child)
+local function FilterServices(Child: Instance): boolean
+	return not not table.find(Settings.Include, Child)
 end
 
-local function encodeInstance(instance: Instance, childFilter: ((Instance) -> boolean)?): EncodedInstance
-	local encoded = {}
-	encoded.Name = instance.Name
-	encoded.ClassName = instance.ClassName
-	encoded.Children = {}
+local function EncodeInstance(Instance: Instance, ChildFilter: ((Instance) -> boolean)?): EncodedInstance
+	local Encoded = {}
+	Encoded.Name = Instance.Name or "EmptyName"
+	Encoded.ClassName = Instance.ClassName or "EmptyClassName"
+	Encoded.Children = {}
 
-	for _, child in instance:GetChildren() do
-		if childFilter and not childFilter(child) then
+	for _, Child in Instance:GetChildren() do
+		if ChildFilter and not ChildFilter(Child) then
 			continue
 		end
 
-		table.insert(encoded.Children, encodeInstance(child))
+		table.insert(Encoded.Children, EncodeInstance(Child))
 	end
 
-	return encoded
+	return Encoded
 end
 
-local function cleanup()
-	for _, connection in pairs(connections) do
-		connection:Disconnect()
+local function CleanUpConnections()
+	local WasConnected = Connected
+	for _, Connection in Connections do
+		Connection:Disconnect()
+	end
+	Connected = false
+	ConnectButton.Icon = "rbxassetid://11115506617"
+
+	if WasConnected then
+		warn("[Luau Language Server] No longer sending DataModel information.")
 	end
-	connected.Value = false
 end
 
-local function sendFullDMInfo(isSilent)
-	local tree = encodeInstance(game, filterServices)
+local function SendDataModelInfo()
+	local Tree = EncodeInstance(game, FilterServices)
 
-	local success, result = pcall(HttpService.RequestAsync, HttpService, {
+	local Success, Result = pcall(HttpService.RequestAsync, HttpService, {
 		Method = "POST" :: "POST",
-		Url = string.format("http://localhost:%s/full", Settings.port),
+		Url = string.format("http://localhost:%s/full", tostring(Settings.Port)),
 		Headers = {
 			["Content-Type"] = "application/json",
 		},
 		Body = HttpService:JSONEncode({
-			tree = tree,
+			tree = Tree,
 		}),
+
 		Compress = Enum.HttpCompression.Gzip,
 	})
 
-	if not success then
-		warn(`[Luau Language Server] Connecting to server failed: {result}`)
-		cleanup()
-	elseif not result.Success then
-		warn(`[Luau Language Server] Sending full DM info failed: {result.StatusCode}: {result.Body}`)
-		cleanup()
+	if not Success then
+		warn(`[Luau Language Server] Connecting to server failed: {Result}`)
+		CleanUpConnections()
+	elseif not Result.Success then
+		warn(`[Luau Language Server] Sending DataModel info failed: {Result.StatusCode}: {Result.Body}`)
+		CleanUpConnections()
 	else
-		connected.Value = true
-		if not isSilent then
-			print(`[Luau Language Server] Successfully sent full DataModel info`)
-		end
+		Connected = true
+		ConnectButton.Icon = "rbxassetid://11116536087"
+		warn("[Luau Language Server] Now sending DataModel information.")
```

Comment:
This warns when any object added/removed, it should only warn on initial connection

Reply:
Agreed, this was a left over fragment from when I was planning to do DebugMode in this pr